routes/auth.routes.js

```diff
@@ -0,0 +1,109 @@
+const User = require("../models/User.model");
+const router = require("express").Router();
+const bcrypt = require("bcryptjs");
+const mongoose = require('mongoose');
+const {isLoggedOut, isLoggedIn} = require("../middleware/route.guard")
+
+
+
+//Sign UP
+
+router.get("/signup", isLoggedOut, (req, res, next) => res.render("auth/signUp"));
+
+router.post("/signup", async (req, res, next) => {
+    const {username, password} = req.body;
+    
+    try {
+        if(!username || !password) {
+            res.render("auth/signUp", {errorMessage: "Please insert an username and password"});
+            return;
+        }
+
+        const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
+        if (!regex.test(password)) {
+        res.status(500).render('auth/signUp', {
+        errorMessage:
+          'Invalid password, password needs to have at least 6 characters and include an uppercase and lowercase character',
+      });
+    }
+        const salt = await bcrypt.genSalt(10);
+        const hashPass = await bcrypt.hash(password, salt)
+        const createdUser = await User.create({username, password: hashPass});
+        const createdUserName = createdUser.username;
+
+        res.redirect(`/login/${createdUser.username}`);
+    } catch (error) {
+        console.log(error);
+        if (error instanceof mongoose.Error.ValidationError){
+            res.status(500).render("auth/signUp", {errorMessage: error.message})
+        }
+        else if (error.code === 11000) {
+        res.status(500).render('auth/signUp', {errorMessage: 'Username already exists'});
+        }
+
+        next(error);
+    }
+})
+
+//log in
+
+router.get("/login/:auth", isLoggedOut, (req, res, next) => {
```

Comment:
on this one, let's just render the auth/login and take the /:auth out

`router.get("/login", (req, res, next) => {
        res.render('auth/login');
});`